test/e2e/lib/browser-manager.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file Manages instance of Playwright Browser and BrowserContext.
+ * @author Edwin Takahashi
+ */
+
+/**
+ * External dependencies
+ */
+import { Browser, BrowserContext, chromium } from 'playwright';
+import config from 'config';
+
+/**
+ * Internal dependencies
+ */
+import type { screenSize, localeCode } from './types';
+
+const browserStartTimeoutMS = 2000;
+
+export let browser: Browser;
+
+/**
+ * Returns the target screen size for tests to run against.
+ *
+ * By default, this function will return 'desktop' as the target.
+ * To specify another screen size, set the BROWSERSIZE environment variable.
+ *
+ * @returns {screenSize} Target screen size.
+ */
+export function getTargetScreenSize(): screenSize {
+	return ! process.env.BROWSERSIZE
+		? 'desktop'
+		: ( process.env.BROWSERSIZE.toLowerCase() as screenSize );
+}
+
+/**
+ * Returns the locale under test.
+ *
+ * By default, this function will return 'en' as the locale.
+ * To set the locale, set the BROWSERLOCALE environment variable.
+ *
+ * @returns {localeCode} Target locale code.
+ */
+export function getTargetLocale(): localeCode {
+	return ! process.env.BROWSERLOCALE ? 'en' : process.env.BROWSERLOCALE.toLowerCase();
+}
+
+/**
+ * Returns a set of screen dimensions in numbers.
+ *
+ * This function takes the output of `getTargetScreenSize` and returns an
+ * object key/value mapping of the screen diemensions represented by
+ * the output.
+ *
+ * @returns {number, number} Object with key/value mapping of screen dimensions.
+ * @throws {Error} If target screen size was not set.
+ */
+export function getScreenDimension(): { width: number; height: number } {
+	switch ( getTargetScreenSize() ) {
+		case 'mobile':
+			return { width: 400, height: 1000 };
+		case 'tablet':
+			return { width: 1024, height: 1000 };
+		case 'desktop':
+			return { width: 1440, height: 1000 };
+		case 'laptop':
+			return { width: 1400, height: 790 };
+		default:
+			throw new Error( 'Unsupported screen size specified.' );
+	}
+}
+
+/**
+ * Returns a new instance of a BrowserContext.
+ *
+ * A BrowserContext represents an isolated environment, akin to incognito mode
+ * inside which a single test suite is run.
+ * BrowserContexts are cheap to create and incur low overhead costs while allowing
+ * for parallelization of test suites.
+ *
+ * @returns {Promise<BrowserContext>} New BrowserContext instance.
+ */
+export async function newBrowserContext(): Promise< BrowserContext > {
+	// If no existing instance of a Browser, then launch a new instance.
+	if ( ! browser ) {
+		browser = await launchBrowser();
+	}
+
+	// Generate a new BrowserContext.
+	return await browser.newContext( {
+		viewport: null, // Do not override window size set in the browser launch parameters.
```

Comment:
I'm wondering if we need this extra abstract layer for the browser instance. Is it intended that we don't want to return the current browser instance? What if we only had the `launchBrowser` method that returns the browser instance and dropped the `quitBrowser` and `newBrowserContext` in favor of the native ones? I guess it would require one extra step to get the instance comparing to the current flow, but at the same time, we're not adding an extra learning curve to the Playwright framework. So, in a spec it could look like this:

```js
import { launchBrowser } from 'browser-manager';

let browser;
let context;

before() {
	browser = await launchBrowser();
	context = await browser.newContext();
}

after() {
	browser.close();
}

describe('Howdy', function() {
	// write some specs
});

```

What do you think, folks?

Reply:
Ah, now I can see that we're sharing the browser instance with Mocha [root hooks](https://github.com/Automattic/wp-calypso/pull/51285/files#diff-5580bcd7dfbb3f4ef83c4299ea79d943d2175fc3fd08b56abdd49808812fbd13R24). Would it make sense then to make the `browser` instance global? (i.e. Launch in `before` and close in `after` root hook)

Reply:
Yes, my intention was to use browser instance in mocha hooks to control browser launch and close. The overall goal I was looking to achieve is to reduce the amount of boilerplate code that test writers need to include in each spec.

With that said however, in my subsequent patches I am discovering that the goal might not be feasible, so this is subject to change.